code/datums/mind.dm

```diff
@@ -125,95 +124,83 @@
 		new_character.key = key		//now transfer the key to link the client to our new body
 	current.update_atom_languages()
 
-///Return the amount of EXP needed to go to the next level. Returns -1 if max level.
-/datum/mind/proc/get_xp_to_level_up(datum/skill/skill_ref)
-	var/exp = skill_experience[skill_ref]
-	if (exp < SKILL_EXP_NOVICE)
-		return SKILL_EXP_NOVICE
-	if (exp < SKILL_EXP_APPRENTICE)
-		return SKILL_EXP_APPRENTICE-SKILL_EXP_NOVICE
-	if (exp < SKILL_EXP_JOURNEYMAN)
-		return SKILL_EXP_JOURNEYMAN-SKILL_EXP_APPRENTICE
-	if (exp < SKILL_EXP_EXPERT)
-		return SKILL_EXP_EXPERT-SKILL_EXP_JOURNEYMAN
-	if (exp < SKILL_EXP_MASTER)
-		return SKILL_EXP_MASTER-SKILL_EXP_EXPERT
-	if (exp < SKILL_EXP_LEGENDARY)
-		return SKILL_EXP_LEGENDARY-SKILL_EXP_MASTER
-	return 0
-
-///Return the amount of EXP needed for previous level
-/datum/mind/proc/xp_req_for_prev_level(datum/skill/skill_ref)
-	var/exp = skill_experience[skill_ref]
-	if (exp > SKILL_EXP_LEGENDARY)
-		return SKILL_EXP_LEGENDARY
-	if (exp > SKILL_EXP_MASTER)
-		return SKILL_EXP_MASTER
-	if (exp > SKILL_EXP_EXPERT)
-		return SKILL_EXP_EXPERT
-	if (exp > SKILL_EXP_JOURNEYMAN)
-		return SKILL_EXP_JOURNEYMAN
-	if (exp > SKILL_EXP_APPRENTICE)
-		return SKILL_EXP_APPRENTICE
-	if (exp > SKILL_EXP_NOVICE)
-		return SKILL_EXP_NOVICE
-	return 0
+/datum/mind/proc/init_known_skills()

```

Comment:
this proc entirely ignores the reason why this was done like this and floods every mob with a list they might not need, don't do this.

Reply:
im gonna flood you with a list you dont need

Reply:
Any reason not to cache these lists?